### PR TITLE
rust_ast::ast => rustc_ast

### PR DESCRIPTION
**Pull request by JulianKnodt**
Rework of #71199 which is a rework #70621

Still working on this but just made the PR to track progress

r? @Dylan-DPC

### CODE REVIEW

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 1]
```diff
@@ -73,7 +73,11 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         &self,
         path_str: &str,
         current_item: &Option<String>,
+<<<<<<< HEAD
         module_id: DefId,
+=======
+        module_id: rustc_ast::NodeId,
+>>>>>>> rust_ast::ast => rustc_ast
```
Comment (jyn514) [inline src/librustdoc/passes/collect_intra_doc_links.rs:80]:
```suggestion
        module_id: DefId,
```

**Comment by jyn514** [comment]
Wow, that is a _lot_ of changed files 😅

File: src/librustdoc/passes/collect_intra_doc_links.rs (hunk 1 above)
Comment (JulianKnodt) [reply src/librustdoc/passes/collect_intra_doc_links.rs:80]: yea I'm getting to these one by one

**Comment by eddyb** [comment]
Please don't r? me on this, I wanted this (for consistency etc.), someone else from @rust-lang/compiler should approve it.

r? @ghost

**Comment by JulianKnodt** [comment]
Is this high priority or just for consistency? If it's slow to review there's no point in me rebasing because it touches so many things the time between merge conflicts is quite low.

File: src/librustc_traits/lowering/mod.rs [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+    ProgramClauseCategory, WellFormed, WhereClause,
+};
+use rustc_middle::ty::query::Providers;
+use rustc_middle::ty::subst::{InternalSubsts, Subst};
+use rustc_middle::ty::{self, List, TyCtxt};
+use rustc_span::symbol::sym;
+
+use std::iter;
+
+crate fn provide(p: &mut Providers<'_>) {
```
Comment (oli-obk) [inline src/librustc_traits/lowering/mod.rs:21]: where did this file come from? I don't think it's necessary for this PR

**Comment by oli-obk** [comment]
@bors p=6 super bitrotty

r=me with the accidentally added file removed.

**Comment by Dylan-DPC** [comment]
@JulianKnodt you need to run tidy once over this and we are good to go

**Comment by matklad** [comment]
cc @mark-i-m , for removing rustc prefix idea

**Comment by mark-i-m** [comment]
I'm pretty torn honestly. I think this change is great, but also look at all that churn! Now imagine doing that literally for the whole compiler.

**Comment by JulianKnodt** [comment]
Is there a way to allow for previous paths to exist but request people update?

**Comment by mark-i-m** [comment]
The only thing I thought of was to use `use ast as rustc_ast;` and so on. So we can change the crate names but continue to use the old names.

**Comment by Dylan-DPC** [comment]
@mark-i-m looks like more PRs to work on, on my list :D

**Pull request merged**